core/controllers/profile_test.py

```diff
@@ -217,3 +216,18 @@ def test_user_disallowing_emails_on_signup(self):
             self.assertEqual(
                 user_services.get_email_preferences(self.EDITOR_ID),
                 {'can_receive_email_updates': False})
+
+class ProfileLinkTests(test_utils.GenericTestBase):
+
+    USERNAME = 'abc123'
+    EMAIL = 'abc123@gmail.com'
+    PROFILE_PIC_URL = '/preferenceshandler/profile_picture_by_username/'
+
+    def test_get_profile_picture_invalid_username(self):
+        response = self.testapp.get(self.PROFILE_PIC_URL + self.USERNAME, expect_errors=True)
```

Comment:
Generally, we prefer string interpolation: '%s%s' % (self.PROFILE_PIC_URL, self.USERNAME)